src/buffer.c

```diff
@@ -124,11 +124,26 @@ buffer_t *init_buffer(const char *filename) {
   // we want to start at the top when user opens up a new file
   buf->current = buf->head;
   // Set current to the first char of line
-  // TODO set current to the first non space char
   buf->current->current = buf->current->head;
 
-  buf->current_row = 0;
   buf->current_char = 0;
+  buf->current_row = 0;
+
+  // go to the first non-empty row
+  if (buf->num_rows > 1) {
+      while (buf->current->line_size == 0) {
+          buf->current = buf->current->next;
```

Comment:
This will potentially cause a `Segmentation fault`. `buf->current` will point to `NULL` if the file consists of three empty lines (line_size is 0).